### PR TITLE
Update warnings.md

### LINKED ISSUE #5417: Docs Changes Needed from dbt-core Issue #9644
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.

### PR DESCRIPTION
**Pull request by mirnawong1**
Link [to staging](https://docs-getdbt-mwowj9p7x-dbt-labs.vercel.app/reference/global-configs/warnings)

Resolves #5417 for v1.8

adds warn.silence to the warnings doc in prep for v1.8

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/global-configs/warnings.md [hunk 1]
```diff
@@ -47,16 +72,34 @@ DBT_WARN_ERROR_OPTIONS='{"include": ["NoNodesForSelectionCriteria"]}' dbt run
 ...
 ```
 
+<VersionBlock lastVersion="1.7">
+
 <File name='profiles.yml'>
```
Comment (runleonarun) [inline website/docs/reference/global-configs/warnings.md:88]:
```suggestion
```

File: website/docs/reference/global-configs/warnings.md [hunk 2]
```diff
@@ -14,16 +14,41 @@ dbt --warn-error run
...
 
 The `WARN_ERROR_OPTIONS` config gives you more granular control over _exactly which types of warnings_ are treated as errors. Warnings that should be treated as errors can be specified through `include` and/or `exclude` parameters. Warning names can be found in [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`).
 
-The `include` parameter can set to `"all"` or `"*"` to treat all warnings as exceptions, or to a list of specific warning names to treat as exceptions. When include is set to `"all"` or `"*"`, the optional `exclude` parameter can be set to exclude specifc warnings from being treated as exceptions.
+The `include` parameter can be set to `"all"` or `"*"` to treat all warnings as exceptions, or to a list of specific warning names to treat as exceptions. When include is set to `"all"` or `"*"`, the optional `exclude` parameter can be set to exclude specific warnings from being treated as exceptions.
```
Comment (runleonarun) [inline website/docs/reference/global-configs/warnings.md]: I think we need to capture part of this with the "1.7 last version" as this is now called "silence"

File: website/docs/reference/global-configs/warnings.md [hunk 3]
```diff
@@ -47,16 +72,33 @@ DBT_WARN_ERROR_OPTIONS='{"include": ["NoNodesForSelectionCriteria"]}' dbt run
 ...
 ```
 
-<File name='profiles.yml'>
+<VersionBlock lastVersion="1.7">
 
+<File name='profiles.yml'>
 ```yaml
```
Comment (runleonarun) [inline website/docs/reference/global-configs/warnings.md:89]:
```suggestion

```yaml
```

File: website/docs/reference/global-configs/warnings.md (hunk 2 above)
Comment (mirnawong1) [reply website/docs/reference/global-configs/warnings.md]: hey @runleonarun i'm not sure i understand - i thought `silence` was a new parameter? so are you saying the entire paragraph should be avail in 1.7 and older right bc include/exclude has changed to error/warn i think?

Comment (runleonarun) [reply website/docs/reference/global-configs/warnings.md]: Hey @mirnawong1 I'm not 100% sure, but I was going by how in the [new example](https://github.com/dbt-labs/docs.getdbt.com/pull/5434/files#diff-92205076f6f62b5d3737e562ef03e8f5c14e4ac9595174f437514528d8df369dR132-R133) we say:

>    error: # Previously called "include"
>    warn: # Previously called "exclude"

It appears as though we're reframing the language for these messages to be more specific: error, warning, silence. Before it took a mental leap to understand the logic (IMO.) If I have time, I will take a look and make some changes to this PR today.

**Comment by runleonarun** [comment]
@QMalcolm and @graciegoheen How should we document this change? This PR versions include/exclude for 1.7 and prior. We versioned warn/error/silence for 1.8 and higher.

Let me know if you'd prefer this handled differently because include/exclude can still be used in 1.8+ (from what I can tell in the PR notes.)

**Comment by QMalcolm** [comment]
> @QMalcolm and @graciegoheen How should we document this change? This PR versions include/exclude for 1.7 and prior. We versioned warn/error/silence for 1.8 and higher.
> 
> Let me know if you'd prefer this handled differently because include/exclude can still be used in 1.8+ (from what I can tell in the PR notes.)

I think moving to the language of `warn/error/silence` for 1.8+ makes sense! The old keys `exclude/include` will still work, and they haven't technically been deprecated (though I can imagine a future where we do deprecate them). However, I think we probably want to be opinionated and get people slowly moved over.

All that said, I would defer to @graciegoheen as I think her perspective is more important here.

File: website/docs/reference/global-configs/warnings.md [hunk 4]
```diff
@@ -12,20 +12,55 @@ Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any ti
 dbt --warn-error run
 ...
 ```
+
 </File>
 
-<VersionBlock firstVersion="1.4">
 
-Converting any and all warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot.
+Converting any warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot.The `WARN_ERROR_OPTIONS` config gives you more granular control over _exactly which types of warnings_ are treated as errors.
```
Comment (runleonarun) [inline website/docs/reference/global-configs/warnings.md]:
```suggestion
Converting any warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot. The `WARN_ERROR_OPTIONS` config gives you more granular control over _exactly which types of warnings_ are treated as errors. 
```

**Review by runleonarun — APPROVED** [review]
This is ready to ship from docs perspective.

**Comment by runleonarun** [comment]
@grace when you're ready to review, here's the [staging link](https://docs-getdbt-com-git-mirnawong1-patch-29-dbt-labs.vercel.app/reference/global-configs/warnings)!

**Comment by graciegoheen** [comment]
I agree with Quigley here - let's use the new `warn/error/silence` language for 1.8+

**Pull request merged**